extensions/cornerstone/src/services/SegmentationService/SegmentationService.ts

```diff
@@ -1132,6 +1134,18 @@ class SegmentationService extends PubSubService {
     }
   };
 
+  private _setDisplaySetIsHydrated(
+    displaySetUID: string,
+    isHydrated: boolean
+  ): void {
+    const {
+      DisplaySetService: displaySetService,
+    } = this.servicesManager.services;
+    const displaySet = displaySetService.getDisplaySetByUID(displaySetUID);
+    displaySet.isHydrated = isHydrated;
+    displaySetService.setDisplaySetMetadataInvalidated(displaySetUID);
```

Comment:
Not sure if this is the best way, but at the very least it will start a discussion.

Reply:
hmmm, I see why you need this, but at the same time we have another listener for this at the OHIFCornerstoneViewport which invalidates the data. Maybe we need another event to decouple this?

Reply:
How about a new `DisplaySetService` event called `DISPLAY_SET_UPDATED`?

Reply:
I'm not sure really, it is kind of the same event we are invalidating a metadata, but at the same time we don't want to hit that other listener in the OHIFCornerstoneViewport, aaaaah
I hate having `DISPLAY_SET_UPDATED` and `DISPLAY_SET_SERIES_METADATA_INVALIDATED`
Let's think till we meet